File: backend/src/TheBlueSky.Flights/Models/Aircraft.cs
using System.ComponentModel.DataAnnotations;
using TheBlueSky.Flights.Enums;

namespace TheBlueSky.Flights.Models
{
    public class Aircraft
    {
        public int AircraftId { get; set; }

        [Required(ErrorMessage = "Flight Owner user ID is required.")]
        public int OwnerUserId { get; set; }

        [Required(ErrorMessage = "Aircraft name is required.")]
        [StringLength(100, ErrorMessage = "Aircraft name cannot exceed 100 characters.")]
        public string AircraftName { get; set; } = default!;

        [Required(ErrorMessage = "Aircraft model is required.")]
        [StringLength(100, ErrorMessage = "Aircraft model cannot exceed 100 characters.")]
        public string AircraftModel { get; set; } = default!;

        public AircraftManufacturer Manufacturer { get; set; } = AircraftManufacturer.Unknown;

        [Required(ErrorMessage = "Number of economy class seats is required.")]
        public int EconomySeats { get; set; }

        [Required(ErrorMessage = "Number of business class seats is required.")]
        public int BusinessSeats { get; set; }

        [Required(ErrorMessage = "Number of first class seats is required.")]
        public int FirstClassSeats { get; set; }

        public bool IsActive { get; set; } = true;
        public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

        public ICollection<AircraftSeat> Seats { get; set; } = new List<AircraftSeat>();
        public ICollection<FlightSchedule> Schedules { get; set; } = new List<FlightSchedule>();


    }
}

File: frontend/TheBlueSky.App/src/types/aircraft.ts
import type { AircraftSeat } from "./aircraftSeat";

export const AIRCRAFT_MANUFACTURERS = ['Boeing', 'Airbus'] as const;

export type AircraftManufacturer = typeof AIRCRAFT_MANUFACTURERS[number];


export type Aircraft = {
  aircraftId: number;
  ownerUserId: string;
  aircraftName: string;
  aircraftModel: string;
  manufacturer: AircraftManufacturer;
  economySeats: number;
  businessSeats: number;
  firstClassSeats: number;
  isActive: boolean;
};

export type NewAircraft = Omit<Aircraft, 'aircraftId' | 'ownerUserId'>;

export type AircraftWithSeats = {
  aircraft: Aircraft;
  seats: AircraftSeat[];
};

export const AIRCRAFT_MODELS = [
  'Boeing 737',
  'Boeing 777',
  'Boeing 787 Dreamliner',
  'Airbus A320',
  'Airbus A350',
  'Airbus A380',
] as const;

export const MODELS_BY_MANUFACTURER: Record<AircraftManufacturer, string[]> = {
  Boeing: ['Boeing 737', 'Boeing 777', 'Boeing 787 Dreamliner'],
  Airbus: ['Airbus A320', 'Airbus A350', 'Airbus A380']
};

export const ALL_AIRCRAFT_MODELS = Object.values(MODELS_BY_MANUFACTURER).flat();
export type AircraftModel = (typeof ALL_AIRCRAFT_MODELS)[number];